custom/icds/management/commands/get_icds_sms_usage.py

```diff
@@ -0,0 +1,102 @@
+import csv
+from corehq.apps.sms.models import SMS
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.utils import is_commcarecase
+from corehq.messaging.smsbackends.icds_nic.models import SQLICDSBackend
+from corehq.util.argparse_types import utc_timestamp
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = ""
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('start_timestamp', type=utc_timestamp)
+        parser.add_argument('end_timestamp', type=utc_timestamp)
+
+    def get_location_id(self, sms):
+        location_id = None
+
+        if sms.couch_recipient:
+            if sms.couch_recipient in self.recipient_id_to_location_id:
+                return self.recipient_id_to_location_id[sms.couch_recipient]
+
+            recipient = sms.recipient
+            if is_commcarecase(recipient):
+                if recipient.type == 'commcare-user':
+                    user = CommCareUser.get_by_user_id(recipient.owner_id)
+                    if user:
+                        location_id = user.location_id
+                else:
+                    location_id = recipient.owner_id
+
+            self.recipient_id_to_location_id[sms.couch_recipient] = location_id
+
+        return location_id
+
+    def get_location(self, sms):
+        location_id = self.get_location_id(sms)
+        if not location_id:
+            return None
+
+        if location_id in self.location_id_to_location:
+            return self.location_id_to_location[location_id]
+
+        location = SQLLocation.by_location_id(location_id)
+        self.location_id_to_location[location_id] = location
+        return location
+
+    def get_state_code(self, location):
+        if not location:
+            return 'unknown'
+
+        if location.location_id in self.location_id_to_state_code:
+            return self.location_id_to_state_code[location.location_id]
+
+        state = location.get_ancestors().filter(location_type__code='state').first()
+        if not state:
+            return 'unknown'
+
+        self.location_id_to_state_code[location.location_id] = state.site_code
+        return state.site_code
+
+    def get_indicator_slug(self, sms):
+        if not isinstance(sms.custom_metadata, dict):
+            return 'unknown'
+
+        return sms.custom_metadata.get('icds_indicator', 'unknown')
```

Comment:
there actually shouldn't be any without custom_metadata since monday (there will be a lot of existing ones before that without it which i'm planning to update in a separate script at some point). the unknown tag here is just a way for me to be alerted to this so i can investigate further if it happens